### DISCUSSION TITLE
I need advice in running a GitHub repo with multiple contributors

### DISCUSSION THREAD

**Discussion by Wollivan** [Wollivan/SquadronLeader#29] [General]
This is the first time I've worked on an Open Source project, so I'm not sure the protocol for accepting merge requests and the like.

Also not sure the protocol for adding authors. Do I maintain total control, or add people who can accept PRs?

I want to make this issue a space to discuss how you think this sort of project should be run :)

**Comment by penyuan** [comment]
> I'm not sure the protocol for accepting merge requests and the like
> Do I maintain total control, or add people who can accept PRs?

I think it's up to you!  😄 I've seen many open source projects and worked with some, if you're up for it I think you can be the only person in charge until if and when it's too much work reviewing and responding to issues and pull requests.

> discuss how you think this sort of project should be run

A few things come to mind: 

1. On [this page](https://github.com/Wollivan/SquadronLeader/community) you can see some recommended good practices for a GitHub repository. I think adding a Code of Conduct and contributing guidelines will make this place more welcoming for contributors. I am happy to submit a pull request with those items for your review in the near future if that helps?

2. For future contributions, it is good practice to ask the author of the pull request to include a [Developer Certificate of Origin](https://elinux.org/Developer_Certificate_Of_Origin). It is pretty simple to do, where the person making the commit adds a line to the commit message saying: `Signed-off-by: Jane Doe <jane.doe@theirdomain.com>`. When they sign off on the commit, they are signifying they agree that their contribution to this repository will be under the CC BY-SA 4.0 license as specified in the `LICENSE` file. I can add this info to the pull request with the contributing guidelines. The reasoning behind all this is explain in [this post](https://drewdevault.com/2021/04/12/DCO.html) by the prominent open source developer Drew DeVault. This will help the project keep track of contributing authors, and also make it easier to maintain the list shown in the `README`.

3. One of the most commonly used code of conduct documents is the [Contributor Covenant](https://www.contributor-covenant.org), which basically asks everyone to be excellent to each other. I'm happy to send this along as a pull request as well.

What do you think?

**Reply by Wollivan** [reply]
Oh gosh, I am so grateful that you're here dude!

If you're truly happy to do these things, I would be incredibly grateful! But please don't feel you have to :)

**Reply by penyuan** [reply]
No worries. I've sent a pull request with a code of conduct. Notice that I've included the Developer Certificate of Origin "signed-off-by" line in the commit message.

I'll try to find a time to suggest a contributor guidelines document, too!

**Reply by penyuan** [reply]
I've just submitted a pull request #45 with a basic contributor guidelines document. Hope it works!
